### PR TITLE
Publish assets immediately and clean some vars

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

**Comment by mmitche** [comment]
Test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1988696&view=results

**Comment by carlossanlop** [comment]
For my own education: Can you please explain the reason/purpose of this change?

**Comment by mmitche** [comment]
> For my own education: Can you please explain the reason/purpose of this change?

It pulls about 10 mins off the official build. In past days, when we ran the Validation stage before we published, we didn't want publishing to run if validation failed, hence the separate stage. Most repos do validation nightly now (since it rarely fails), This means that we have a separate stage after publishing to BAR that just calls into darc to launch publishing. This stage has overhead (machine allocation, cloning, etc.).

Since we don't need to run validation, we can just have the Publish To Build Asset Registry job immediately call darc to publish the build.

File: eng/pipelines/runtime-official.yml [hunk 1]
```diff
@@ -27,11 +27,6 @@ variables:
 # TODO: (Consolidation) Switch away from old signing/validation variables from former Core-Setup. https://github.com/dotnet/runtime/issues/1027
 - name: TeamName
   value: dotnet-core-acquisition
-# Set the target blob feed for package publish during official and validation builds.
-- name: _DotNetArtifactsCategory
-  value: .NETCore
-- name: _DotNetValidationArtifactsCategory
-  value: .NETCoreValidation
```
Comment (carlossanlop) [inline eng/pipelines/runtime-official.yml:34]: Are these the lines that were indicating validation to be done separately?

**Review by carlossanlop — APPROVED** [review]

File: eng/pipelines/runtime-official.yml (hunk 1 above)
Comment (mmitche) [reply eng/pipelines/runtime-official.yml:34]: They're old parameters that no longer have any use.

**Pull request merged**